o2a/o2a_libs/property_utils.py

```diff
@@ -39,6 +40,12 @@ def __init__(
         self.config: Dict[str, str] = config or {}
         self.action_node_properties: Dict[str, str] = action_node_properties or {}
 
+    @property
+    def xml_escaped(self):
```

Comment:
Property should not modify the state of the object. Property when it needs to modify the object should make copies of the object. Otherwise, understanding the object's condition can be difficult. It is also possible accidental corruption of the object.

Reply:
Agree. Missed that one.

Reply:
Thanks. I implemented a version with copying the object in 99797c5a96fce013cf362e2ec8c0ea00ccd383b6. Also added a unit test.
Let me know if you think this implementation is ok.

Reply:
LGTM.